src/main/start-main-application/lens-window/hide-all-windows/close-all-windows.injectable.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import { lensWindowInjectionToken } from "../application-window/lens-window-injection-token";
+
+const closeAllWindowsInjectable = getInjectable({
+  id: "close-all-windows",
+
+  instantiate: (di) => () => {
+    const lensWindows = di.injectMany(lensWindowInjectionToken);
+
+    lensWindows.forEach((lensWindow) => {
```

Comment:
question: Why the preference of this over `for...of`?

Reply:
`for...of` is something that you cannot compose. 

Imagine:


```typescript
for (const item of items) {
  doSomething(item);
}
```

vs.

```typescript
someItems.forEach(item) => doSomething(item);
```


Now let's say we want to filter and sort items before we do it.

```typescript
for (const item of items.filter(x => !!x.something).sort()) {
  doSomething(item);
}
```

vs.

```typescript
someItems.filter(x => !!x.something).sort().forEach(item) => doSomething(item);
```

or even

```typescript
pipeline(
  someItems,
  filter(x => !!x.something),
  forEach(doSomething)
)
```

Which one leads to cleaner code? :)